enterprise/views.py

```diff
@@ -730,6 +720,24 @@ def get(self, request):  # pylint: disable=too-many-statements
                 )
                 return render_page_with_error_code_message(request, context_data, error_code, log_message)
 
+            # In the event that a learner did enroll into audit from B2C and they consented on the data sharing consent
+            # page we want to upgrade their audit enrollment into verified when they view the course in the learner
+            # portal and hit this endpoint again
+            upgrade_to_license_enrollment = consent_record is not None and consent_record.granted and license_uuid
```

Comment:
hmm, trying to understand the implication here if someone declined consent... we assumed before that they would just get asked again but i think it's worth testing.

Reply:
Well consent_record.granted would be false if they declined so they'll still have to give consent, but I will manually verify that's the case.